mp/src/game/shared/momentum/mom_rocket.cpp

```diff
@@ -195,8 +206,39 @@ void CMomRocket::Explode(trace_t *pTrace, CBaseEntity *pOther)
     float flDamage = GetDamage();
     float flRadius = GetRadius();
 
-    // Create explosion effect with no damage
-    ExplosionCreate(vecOrigin, GetAbsAngles(), pOwner, flDamage, flRadius, false);
+    static ConVarRef mom_rj_sounds("mom_rj_sounds");
+    CPVSFilter filter(vecOrigin);
+    int EntIndex = 0;
+    EntIndex = pOther->entindex();
+    CWeaponID m_hWeaponID = WEAPON_ROCKETLAUNCHER;
+
+    // Create explosion effect with no damage depending on the particle cvar setting
+    switch (mom_rj_particles.GetInt())
+    {
+    case 0:
+        TE_TFExplosion(filter, 0.0f, vecOrigin, pTrace->plane.normal, m_hWeaponID, EntIndex);
+        break;
+    case 1:
+        if (mom_rj_sounds.GetInt() == 0)
+        {
+            // Silent explosion
+            ExplosionCreate(vecOrigin, GetAbsAngles(), pOwner, flDamage, flRadius, false, 0.0f, false, true);
+        }
+        else if (mom_rj_sounds.GetInt() == 1)
+        {
+            ExplosionCreate(vecOrigin, GetAbsAngles(), pOwner, flDamage, flRadius, false);
+        }
+        else if (mom_rj_sounds.GetInt() == 2)
+        {
+            // Small hack: If TF2 sounds are selected but Momentum particles, then use silent standard explosion effect and play TF2 explosion sound
+            ExplosionCreate(vecOrigin, GetAbsAngles(), pOwner, flDamage, flRadius, false, 0.0f, false, true);
+            CBaseEntity::EmitSound(filter, SOUND_FROM_WORLD, "BaseExplosionEffect.SoundTF2", &vecOrigin);
+        }
+        break;
+    case 2:
+        TE_TFExplosion(filter, 0.0f, vecOrigin, pTrace->plane.normal, m_hWeaponID, EntIndex);
+        break;
+    }
```

Comment:
Pull all of this out into a `CreateRocketExplosionEffect()` method, it's crowding the `Explode` one.